ProxySQL reset connection if it starts with COM_QUIT

- [x] ProxySQL version: 2.0.15
- [x] OS version: Official Docker image: proxysql/proxysql:2.0.15


Description
-----

I'm trying make zero-downtime migration from MASTER (MariaDB) to BALANCER (ProxySQL)
Now i have installed haproxy on my machines and all local applications use that address.
Now HAProxy balance connection between two MASTER IPs (main and reserved).

I have read-only replica (MariaDB) and want split traffic. For that i was add balancer (ProxySQL) address to HAProxy config.
ProxySQL address tagged as unhealthy and traffic not routed to him.

 **I'm investigate that issue and here is results.**

Issue
----
HAProxy check mysql health status by sending `COM_QUIT` to the server and wait `OK` and close connection ([see here](https://github.com/haproxy/haproxy/blob/cd03be73d59ad2c5d9ad43b28a1b0fda9f32526c/src/tcpcheck.c#L4532))
This behaviour described in MySQL [protocol official documentation](https://dev.mysql.com/doc/internals/en/com-quit.html)

MariaDB flow (got `OK`, all fine)

![изображение](https://user-images.githubusercontent.com/735150/111038928-403cb280-844d-11eb-8a62-78b6d36b4e78.png)


ProxySQL flow (no `OK` response, TCP RST only):

![изображение](https://user-images.githubusercontent.com/735150/111039138-5139f380-844e-11eb-9676-1e30854f0c8e.png)


Error log
------

ProxySQL print this errors for each HAProxy connection attempt:

2021-03-13 17:42:58 mysql_data_stream.cpp:411:check_data_flow(): [ERROR] Session=0x7f53a545b800, DataStream=0x7f53a5448c00 -- Data at both ends of a MySQL data stream: IN <1 bytes 0 packets> , OUT <1 bytes 0 packets>
2021-03-13 17:42:58 MySQL_Thread.cpp:4439:process_all_sessions(): [WARNING] Closing unhealthy client connection 94.19x.xx.xx:51320

--- a/lib/MySQL_Thread.cpp
+++ b/lib/MySQL_Thread.cpp
@@ -5245,6 +5245,12 @@ void MySQL_Thread::listener_handle_new_connection(MySQL_Data_Stream *myds, unsig
 		ioctl_FIONBIO(sess->client_myds->fd, 1);
 		mypolls.add(POLLIN|POLLOUT, sess->client_myds->fd, sess->client_myds, curtime);
 		proxy_debug(PROXY_DEBUG_NET,1,"Session=%p -- Adding client FD %d\n", sess, sess->client_myds->fd);
+
+		// we now enforce sending the 'initial handshake packet' as soon as it's generated. This
+		// is done to prevent situations in which a client sends a packet *before* receiving
+		// this 'initial handshake', leading to invalid state in dataflow, since it will be
+		// data in both ends of the datastream. For more details see #3342.
+		sess->writeout();
 	} else {
 		free(addr);
 		// if we arrive here, accept() failed

